rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/CoordinatorBookKeeper.java

```diff
@@ -84,6 +162,20 @@ public void handleHeartbeat(String workerHostname, HeartbeatStatus heartbeatStat
     }
   }
 
+  @Override
+  public Map<String, NodeState> getClusterNodes()
+  {
+    return getClusterManager().getNodes();
+  }
+
+  @Override
+  public String getOwnerNodeForPath(String remotePathKey)
+  {
+    Map<String, NodeState> nodesMap = getClusterNodes();
```

Comment:
1. Move getOwnerNodeForPath to BookKeeper.java
2. Make getClusterNodes abstract method in BookKeeper.java
3. Remove getOwnerNodeForPath from WorkerBookKeeper and CoordinatorBookKeeper